### DISCUSSION TITLE
Migrate from EMQx V4 to V5.7.1

### DISCUSSION THREAD

**Discussion by Libra23986** [emqx/emqx#13439] [Q&A] [answered]
Hi, I would like to ask about the emqx.conf, currently I need to update the configuration for it, but there are several configuration items that I couldn't find the replacement, plz need your help on this:
node {
  #async_threads = 32
  #max_ports = 1048576
  global_gc_interval = "5m"
  #fullsweep_after = 1000
  #crash_dump = "log/crash.dump"
  #dist_net_ticktime = 60
  #dist_listen_min = 6369
  #dist_listen_max = 6369
}

rpc {
  #tcp_client_port = 5369
}

plugins {
  #etc_dir = "etc/plugins/"
  #expand_plugins_dir = "plugins/"
}

broker {
  sys_interval = 1m
  sys_heartbeat = 30s
}

Where is zone configuration is configured? Any documents related to change from listener to listeners

**Comment by zmstone** [comment]
Hi @Libra23986 
In v5, zone concept is demoted, for most of the use cases, the implicitly existing `default` zone should be enough.
However, if you indeed need to create more fine-grained control of the mqtt parameters etc, you can read more here:
https://docs.emqx.com/en/enterprise/latest/configuration/configuration.html#zone-override

for config layout in general, you can find the config manual here:
https://docs.emqx.com/en/enterprise/v5.7.1/hocon/

**Reply by Libra23986** [reply]
Thanks for your suggestion, so how's about the other configuration like plugins ..., those one that I listed out.

**Reply by zmstone** [reply]
incompatible changes are documented here:
https://docs.emqx.com/en/enterprise/latest/changes/breaking-changes-5.1.0.html
On the left side navigation bar, you can find more articles about compatibility changes.

**Reply by Libra23986** [reply]
Thanks you, let me check. some of them were deprecated.

**Reply by Libra23986** [reply]
Hi @zmstone , 

I tried to update those configurations also, but unable to find the replacement, are they deprecated? Can you help to advise?
node {
  #dist_listen_min = 6369
  #dist_listen_max = 6369
}

**Reply by zmstone** [reply]
customized dist port is no longer supported.
See port mapping doc here: https://docs.emqx.com/en/enterprise/latest/deploy/cluster/security.html#port-mapping

**Comment by Libra23986** [comment]
@zmstone , thanks for your guideline.
Just want to ask more for those ones:
`node.async_threads`
`node.fullsweep_after`
`plugins.etc_dir`
`plugins.expand_plugins_dir`

**Reply by id** [reply, marked as answer]
@Libra23986 you can ignore these settings
